rotkehlchen/chain/ethereum/modules/cowswap/accountant.py

```diff
@@ -0,0 +1,46 @@
+import logging
+from typing import TYPE_CHECKING
+
+from rotkehlchen.accounting.structures.base import get_tx_event_type_identifier
+from rotkehlchen.accounting.structures.types import HistoryEventSubType, HistoryEventType
+from rotkehlchen.chain.evm.accounting.interfaces import ModuleAccountantInterface
+from rotkehlchen.chain.evm.accounting.structures import TxAccountingTreatment, TxEventSettings
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+
+from .constants import CPT_COWSWAP
+
+if TYPE_CHECKING:
+    from rotkehlchen.accounting.pot import AccountingPot
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+class CowswapAccountant(ModuleAccountantInterface):
+    def event_settings(self, pot: 'AccountingPot') -> dict[str, TxEventSettings]:  # pylint: disable=unused-argument  # noqa: E501
+        """Being defined at function call time is fine since this function is called only once"""
+        return {
+            get_tx_event_type_identifier(HistoryEventType.TRADE, HistoryEventSubType.SPEND, CPT_COWSWAP): TxEventSettings(  # noqa: E501
+                taxable=True,
+                count_entire_amount_spend=False,
+                count_cost_basis_pnl=True,
+                method='spend',
+                accounting_treatment=TxAccountingTreatment.SWAP,
+            ),
+            get_tx_event_type_identifier(HistoryEventType.DEPOSIT, HistoryEventSubType.DEPOSIT_ASSET, CPT_COWSWAP): TxEventSettings(  # noqa: E501
+                taxable=False,
+                count_entire_amount_spend=False,
+                count_cost_basis_pnl=False,
+                method='spend',
+                # Skipping since this event is informational and should not affect asset balance
+                accounting_treatment=TxAccountingTreatment.SKIP,
```

Comment:
Technically - yes. In reality when orders expire they get automatically refunded by this address https://etherscan.io/address/0x0214ae5fd178986fa18ff792e0b995dc6a78cd56

Do you have any better idea on how to handle this better, btw? Since we have to have the spend and the receive event together in accounting when processing swaps I have not come up with a better idea.